bci_main.py

```diff
@@ -58,11 +59,17 @@ def execute_task(task_type: dict, parameters: dict, save_folder: str) -> bool:
         save_folder (str): path to save folder
     """
 
+    exp_type = ExperimentType(task_type['exp_type'])
+
+    # Initialize Display Window
+    display = init_display_window(parameters)
```

Comment:
Hm, interesting. I have not encountered this yet. I did some testing and tried to throw exceptions at various places in the code and the window seemed to cleanup okay. Do you have a more specific example of when you have had this issue? 

The main problems I was trying to solve here was first of all, to feel more responsive during loading, since there is a lot of initialization that happens and it is easy to think that the click did not happen correctly. Secondly, I wanted to fail immediately in the main thread if the user selected an invalid experiment type, rather than waiting for all that loading to occur and have to clean it all up (which requires waiting for multiple threads to stop).

My inclination is to leave it as is for now and then fix the initialization issues as they come, but if you think that would make the system harder to manage let me know and I can revert back.